enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  ## my_select method
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  ## my_all method
+  def my_all
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
```

Comment:
- [ ] enumerables #my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [ ] enumerables #my_all? when no block or argument is given returns false when one of the collection members are false or nil
- [ ] enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer
- [ ] enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
- [ ] enumerables #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class
- [ ] enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] enumerables #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- [ ] enumerables #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
- [ ] enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern